clang/lib/Driver/Driver.cpp

```diff
@@ -3766,6 +3766,20 @@ class OffloadingActionBuilder final {
         return SYCLDeviceOnly ? ABRT_Ignore_Host : ABRT_Success;
       }
 
+      if (CurPhase == phases::Preprocess) {
```

Comment:
I'd maybe suggest moving this block above the `phases::Compile` one to adhere to this "Frontend to backend" policy the rest of the function mimics.